src/dbnode/persist/fs/retriever.go

```diff
@@ -564,6 +569,11 @@ func (r *blockRetriever) streamRequest(
 	startTime time.Time,
 	nsCtx namespace.Context,
 ) (bool, error) {
+	req.resultWg.Add(1)
```

Comment:
any significant in moving this line up?

Reply:
yea since the function returns earlier now, it might block a reader. while a caller shouldn't be waiting on a result that returned an error, we saw it happen when testing.